terra/contracts/token-bridge/src/contract.rs

```diff
@@ -60,90 +61,9 @@ pub enum TransferType<A> {
-    // wrapped assets to 767. However, an upgrade has been attempted to do just that:
-    // https://finder.terra.money/classic/tx/FE39E9549770F59E2AAA1C6B0B86DDF36A4C56CED0CFB0CA4C9D4CC9FBE1E5BA
-    // and it failed with a cryptic 'out of gas error' (notice that the
-    // 'gasWanted' is larger than 'gasUsed', which should never happen. The
-    // process also fails with the same error in simulation, which should
-    // absolutely never happen). When attempting the update locally, it proceeds
-    // without problems, so the current theory is that on mainnet the process
-    // sends too many messages and somehow overflows the gas counter (although
-    // that should result in a different message).
-
-    // The following code doesn't succeed on mainnet (but does locally):
-    /*
-    let messages = migrate_wrapped_assets(deps, env, new_code_id)?;
-    let count = messages.len();
-
-    Ok(Response::new()
-        .add_messages(messages)
-        .add_attribute("migrate", "upgrade cw20 wrappers")
-        .add_attribute("count", count.to_string()))
-    */
-
-    // tldr: The migration strategy currently is to just upgrade the wrapped
-    // asset id variable, so at least new contracts are instantiated from the
-    // correct code id.
-
-    // Just to be safe, we add the above VAA to the replay protection cache so
-    // it can't be reused later if the gas parameters of terra classic change.
-    let data: Binary = Binary::from_base64("AQAAAAINAJG2fJDyk1xTNYrByMq/IyMPXCSLw7uPBviki4lxrp7VMid+Zg5maFTA7L/hr0wjqg2HvStl1ThaVgs+gtP7r88BA8C6uVKxqz+T35SPMuO5EfRXVppuKk4kgC8AaeMXiT/mKN+G4Ywhq7BhtDfFWBT8r/CONWcr4ukcR4WI9SLo6egABJWNvBNzS87mGaYxNIbh9IJRz8IWt/bUbEM01IHxYDLlcKyF+F5XGcbgtolO9BH0SqJiCSuZf0p2Q3i6JbQPqZQABljhQuTzWBrfh+e/tKwo4woE3I7m7ypwgixDiYu5M7NmE1yRlK/1pvqZDKEQQuwRNeF2bB9Feig95jNhyOl8m4AACOQWt3zZAgFON5iebfkfbGR2j0ct0aYCfZgDLS3Tbs+Zb65tN5otY/JB+276JBsV+neyerpYjEv40aw+myj/9hwACkUWUVQ+fiU3PbQI35LLv1bVEUxMnjNsNnVAx2Zcx3trE/c05ODKWLmcgZC7kYR1TXEUgJDEyR7aRx6lh0fYTNMBC5XgNXIgV0QMepMo+rqEO8GSLje0KgiBuESUwerfb0+gbt3HMiCO+4qnGQLHR9H2e0ENo6i+CVd5ArNbUJOZK7IBDMiWbmpGwiTqPKxRDoRjpcDSCVxBU38JkQhA28mJ/swlFisffm8P9mUjFXm75bWqjxWquhLotlD4wHD8n6WKt8YBDThD1TsiQtGY7PpCV1fXc8JGtnxatZAqBiflM9Byo2hEHpVGHoaB2xC+b5ZPnU7SQb/5yXRwxuNGnB07+8P5rUcBDnu+FszGdOsFKLot3vZj8m6dznXAbxi8VHBjf+DZg0b4VUeAEJO5y4iAHEJf+vzXyaHDuw6uaZy/nE6Cmh2G2WoBDxHFcHxwEmPPImzO3qd/ccUiLE8WLVD7awpYQzuSRhbPbBfluVx7ayBD4tzojqQmO0ST+2FWVjZLYIc2o8a6cwsBEfMoHr6CDGnRx/JYvffw1TCQN4bc736cjPvrto++VciZNXE7/EEWyM8oFiDUDX58COwoZacwkE9ujwlPX9MutOEBEqVTBk6s5r136OH6OXNi7SmCJiMwiN+cNghtIKvXt+dvONA56YZrzsTu2N0skC9xgykjGyxGQBmKTI18I/3r9cMAAAAAAORtswQAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEXKpMi7kC0wYgAAAAAAAAAAAAAAAAAAAAAAAAAAAAVG9rZW5CcmlkZ2UCAAMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAXHw==")?;
-    let vaa = parse_vaa(deps.as_ref(), env.block.time.seconds(), &data)?;
-
-    vaa_archive_add(deps.storage, vaa.hash.as_slice())?;
```

Comment:
let's double check that this VAA is indeed replay protected on mainnet

Reply:
Looks like it is.

I started with

```
worm parse AQAAAAINAJG2fJDyk1xTNYrByMq/IyMPXCSLw7uPBviki4lxrp7VMid+Zg5maFTA7L/hr0wjqg2HvStl1ThaVgs+gtP7r88BA8C6uVKxqz+T35SPMuO5EfRXVppuKk4kgC8AaeMXiT/mKN+G4Ywhq7BhtDfFWBT8r/CONWcr4ukcR4WI9SLo6egABJWNvBNzS87mGaYxNIbh9IJRz8IWt/bUbEM01IHxYDLlcKyF+F5XGcbgtolO9BH0SqJiCSuZf0p2Q3i6JbQPqZQABljhQuTzWBrfh+e/tKwo4woE3I7m7ypwgixDiYu5M7NmE1yRlK/1pvqZDKEQQuwRNeF2bB9Feig95jNhyOl8m4AACOQWt3zZAgFON5iebfkfbGR2j0ct0aYCfZgDLS3Tbs+Zb65tN5otY/JB+276JBsV+neyerpYjEv40aw+myj/9hwACkUWUVQ+fiU3PbQI35LLv1bVEUxMnjNsNnVAx2Zcx3trE/c05ODKWLmcgZC7kYR1TXEUgJDEyR7aRx6lh0fYTNMBC5XgNXIgV0QMepMo+rqEO8GSLje0KgiBuESUwerfb0+gbt3HMiCO+4qnGQLHR9H2e0ENo6i+CVd5ArNbUJOZK7IBDMiWbmpGwiTqPKxRDoRjpcDSCVxBU38JkQhA28mJ/swlFisffm8P9mUjFXm75bWqjxWquhLotlD4wHD8n6WKt8YBDThD1TsiQtGY7PpCV1fXc8JGtnxatZAqBiflM9Byo2hEHpVGHoaB2xC+b5ZPnU7SQb/5yXRwxuNGnB07+8P5rUcBDnu+FszGdOsFKLot3vZj8m6dznXAbxi8VHBjf+DZg0b4VUeAEJO5y4iAHEJf+vzXyaHDuw6uaZy/nE6Cmh2G2WoBDxHFcHxwEmPPImzO3qd/ccUiLE8WLVD7awpYQzuSRhbPbBfluVx7ayBD4tzojqQmO0ST+2FWVjZLYIc2o8a6cwsBEfMoHr6CDGnRx/JYvffw1TCQN4bc736cjPvrto++VciZNXE7/EEWyM8oFiDUDX58COwoZacwkE9ujwlPX9MutOEBEqVTBk6s5r136OH6OXNi7SmCJiMwiN+cNghtIKvXt+dvONA56YZrzsTu2N0skC9xgykjGyxGQBmKTI18I/3r9cMAAAAAAORtswQAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEXKpMi7kC0wYgAAAAAAAAAAAAAAAAAAAAAAAAAAAAVG9rZW5CcmlkZ2UCAAMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAXHw==
```

which results in a digest of

```
0xaf6a8f1929882dd02a5f8412be273c36d9ce9818d0181f475abbe63448718be7
```

If we drop into `node` and concatenate that with the key used by `vaa_archive_add`

```
> Buffer.from("\x00\fguardian_set",'ascii').toString('hex') + "af6a8f1929882dd02a5f8412be273c36d9ce9818d0181f475abbe63448718be7".toUpperCase()
'000c677561726469616e5f736574AF6A8F1929882DD02A5F8412BE273C36D9CE9818D0181F475ABBE63448718BE7'
```

We can then get the that from the public endpoint by paginating on the key just before it.

```
> Buffer.from("000C677561726469616E5F736574AF6A8F1929882DD02A5F8412BE273C36D9CE9818D0181F475ABBE63448718BE6",'hex').toString('base64')
'AAxndWFyZGlhbl9zZXSvao8ZKYgt0CpfhBK+Jzw22c6YGNAYH0dau+Y0SHGL5g=='
```

https://terra-classic-lcd.publicnode.com/cosmwasm/wasm/v1/contract/terra10nmmwe8r3g99a9newtqa7a75xfgs2e8z87r2sf/state?pagination.key=AAxndWFyZGlhbl9zZXSvao8ZKYgt0CpfhBK%2BJzw22c6YGNAYH0dau%2BY0SHGL5g%3D%3D&pagination.limit=1

```
{
  "models": [
    {
      "key": "000C677561726469616E5F736574AF6A8F1929882DD02A5F8412BE273C36D9CE9818D0181F475ABBE63448718BE7",
      "value": "dHJ1ZQ=="
    }
  ],
  "pagination": {
    "next_key": "AAxndWFyZGlhbl9zZXSvauH1bEySKG8sKxiUPHgmTHgHDM93NOkOkKdjxrVIRQ==",
    "total": "0"
  }
}
```

and again `node` can confirm that the value is `true`

```
> Buffer.from("dHJ1ZQ==",'base64').toString('ascii')
'true'
```